src/main/java/org/checkstyle/autofix/recipe/FinalLocalVariable.java

```diff
@@ -56,17 +60,84 @@ public String getDescription() {
 
     @Override
     public TreeVisitor<?, ExecutionContext> getVisitor() {
-        return new LocalVariableVisitor();
+        return new JavaIsoVisitor<>() {
+            @Override
+            public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu,
+                                                          ExecutionContext ctx) {
+                return new LocalVariableVisitor()
+                        .visitCompilationUnit(new MarkViolationVisitor()
+                                .visitCompilationUnit(cu, ctx), ctx);
+            }
+        };
     }
 
-    private final class LocalVariableVisitor extends JavaIsoVisitor<ExecutionContext> {
+    private final class MarkViolationVisitor extends JavaIsoVisitor<ExecutionContext> {
```

Comment:
We definitely need more documentation around these two visitors. The logic looks cumbersome, and I think we should provide some details on how it works